streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -674,6 +676,8 @@ class StreamingContext private[streaming] (
           logWarning("StreamingContext has already been stopped")
         case ACTIVE =>
           scheduler.stop(stopGracefully)
+          // De-registering Streaming Metrics of the StreamingContext
+          env.metricsSystem.removeSource(streamingSource)
```

Comment:
I think here remove the source in `ACTIVE` state may introduce some corner case problem. For example, since metrics source is added when StreamingContext's state is `INITIALIZED`, if we met exception at this point, we will never change the state into `ACTIVE`, so metrics source cannot be removed, since you only assume the state is `ACTIVE` to remove the source. What do you think?